#NR March 2018 - this is my analysis of FS1 data for final figures
#note that most of this was generated by Jules previously, I'm just recombining things

library(tidyverse)
library(reshape2)
library(dplyr)

setwd("/Users/nicolericker/Documents/Pig experiments/FY2016/")
meta<-read.csv(file="FS1.metadata.with.weights.csv", stringsAsFactors = TRUE, header=TRUE)
meta<-meta[,-c(13:14)]

oxy<-read.csv(file="oxytet-values.csv", header = TRUE, stringsAsFactors = FALSE)
colnames(oxy)[2]<-"Day"
oxy$Day<-sub("D", "", oxy$Day)

#need to replace the NF values, empty cells, and < 1.0 values
oxy$Serum<-sub("NF", "0", oxy$Serum)
oxy$Serum<-sub("^$", "0", oxy$Serum)
oxy$Serum<-sub("<", "", oxy$Serum)

oxy$Nasal<-sub("NF", "0", oxy$Nasal)
oxy$Nasal<-sub("^$", "0", oxy$Nasal)
oxy$Nasal<-sub("<", "", oxy$Nasal)

oxy$Ileum<-sub("NF", "0", oxy$Ileum)
oxy$Ileum<-sub("^$", "0", oxy$Ileum)
oxy$Ileum<-sub("<", "", oxy$Ileum)

oxy$Fecal<-sub("NF", "0", oxy$Fecal)
oxy$Fecal<-sub("^$", "0", oxy$Fecal)
oxy$Fecal<-sub("<", "", oxy$Fecal)

#this includes all samples (later included only the positive ones)
all_oxy<-full_join(oxy, meta, by = "Pignum")
colnames(all_oxy)[2]<-"Day.oxy"
colnames(all_oxy)[12]<-"Day"

all_oxy$Serum<-as.numeric(all_oxy$Serum)
all_oxy$Ileum<-as.numeric(all_oxy$Ileum)
all_oxy$Nasal<-as.numeric(all_oxy$Nasal)
all_oxy$Fecal<-as.numeric(all_oxy$Fecal)

all_oxy$Treatment<-factor(all_oxy$Treatment, c("Control", "Inject", "InFeed"))

oxy_present<-oxy %>% filter(oxy$Serum > 0 | oxy$Nasal >0 | oxy$Ileum > 0 | oxy$Fecal >0)
weight_oxy <- inner_join(oxy_present, meta, by = "Pignum", copy=FALSE) 
colnames(weight_oxy)[2]<-"Day.oxy"
colnames(weight_oxy)[12]<-"Day"

weight_oxy$Serum<-as.numeric(weight_oxy$Serum)
weight_oxy$Ileum<-as.numeric(weight_oxy$Ileum)
weight_oxy$Nasal<-as.numeric(weight_oxy$Nasal)
weight_oxy$Fecal<-as.numeric(weight_oxy$Fecal)

day4 <- subset(weight_oxy, Day=="4")

day4 %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Relative.weight, y=Serum, colour=Treatment)) + geom_point() + geom_line(aes(group=Treatment, colour=Treatment))

day4 %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Relative.weight, y=Serum, colour=Treatment)) + geom_point(size=4) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm')


day4 %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Weight.necropsy, y=Fecal, colour=Treatment)) + geom_point(size=4) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm')


day4 %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Weight.necropsy, y=Serum, colour=Treatment)) + geom_point(size=3.5) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') + 
  ggtitle("Weight at necropsy vs. Oxytet in Serum", subtitle = "Only the Day 4 Necropsy Pigs included")


day4 %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Weight.necropsy, y=Ileum, colour=Treatment)) + geom_point(size=4) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm')
day4 %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Weight.necropsy, y=Nasal, colour=Treatment)) + geom_point(size=4) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm')
weight_oxy$Day.oxy<-as.numeric(weight_oxy$Day.oxy)  #I don't think this was necessary actually


weight_oxy %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Weight, y=Serum, colour=Treatment)) + geom_point() + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  ggtitle("Oxytet concetration in serum at Day 4 plotted against weight")
#used this figure in the paper draft (April 4, 2018)
all_oxy %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Weight, y=Serum, colour=Treatment)) + geom_point() + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  ggtitle("Oxytet concetration in serum at Day 4 plotted against weight")

weight_oxy %>% filter(Day.oxy==1) %>%
  ggplot(aes(x=Weight, y=Serum, colour=Treatment)) + geom_point() + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  ggtitle("Oxytet concentration in serum at Day 1 plotted against weight") + xlab('Weight on day 0 (kg)') + ylab('Serum concentration (ng/mL)')
#used this figure in the paper draft (April 4, 2018)
all_oxy %>% filter(Day.oxy==1) %>%
  ggplot(aes(x=Weight, y=Serum, colour=Treatment)) + geom_point(size=0.7, alpha=0.5) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  ggtitle("Serum concentration at Day 1 plotted against weight") + xlab('Weight on day 0 (kg)') + ylab('Serum concentration (ng/mL)')

#removing the title for the poster and correcting to plasma instead of serum
all_oxy %>% filter(Day.oxy==1) %>%
  ggplot(aes(x=Weight, y=Serum, colour=Treatment)) + geom_point(size=0.7, alpha=0.5) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  xlab('Weight on day 0 (kg)') + ylab('Plasma concentration (ng/mL)')

dev.print(tiff, "oxy_plasma-by-weight.tiff", res=800, height=2.5, width=4, units="in")
all_oxy_d4 <- all_oxy %>% filter(Day.oxy==4)
###############


all_oxy %>% filter(Day.oxy==1) %>%
  ggplot(aes(x=Relative.weight, y=Serum, colour=Treatment)) + geom_point() + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  ggtitle("Oxytet concentration in serum on Day 1 plotted against relative weight") + xlab('Relative weight compared to others at necropsy') + ylab('Serum concentration (ng/mL)')

#this one looks a little different than the weight one and may be good to include
all_oxy %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Relative.weight, y=Serum, colour=Treatment)) + geom_point() + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  ggtitle("Oxytet concentration in serum on Day 1 plotted against relative weight") + xlab('Relative weight compared to others at necropsy') + ylab('Serum concentration (ng/mL)')

weight_oxy %>% 
  ggplot(aes(x=Weight, y=Serum, colour=Treatment)) + geom_point() + facet_grid(~Day.oxy) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  ggtitle("Oxytet concetration in serum at Day 4 plotted against weight")

weight_oxy %>% filter(Day.oxy==4) %>%
  ggplot(aes(x=Daily.gain, y=Serum, colour=Treatment)) + geom_point() + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm') +
  ggtitle("Oxytet concentration in serum at Day 4 plotted against daily gain") + xlab('Average daily gain (kg)') + ylab('Serum concentration (ng/mL)')

weight_oxy$Day.oxy<-as.numeric(weight_oxy$Day.oxy)
weight_oxy %>% filter(Treatment=="Inject") %>% 
  ggplot(aes(x=Day.oxy, y=Serum, colour=Pignum, group=Pignum)) + geom_smooth(method='lm') +
  ggtitle("Oxytet concentration in serum over time for inject") + xlab('Time (d)') + ylab('Oxytet concentration (ng/mL)')

weight_oxy %>% filter(Treatment=="InFeed") %>% filter(Day.oxy > 0) %>%
  ggplot(aes(x=Day.oxy, y=Fecal, colour=Pignum, group=Pignum)) + geom_smooth(method='lm') +
  ggtitle("Oxytet concentration in feces over time for InFeed") + xlab('Time (d)') + ylab('Oxytet concentration (ng/mL)')

weight_oxy %>%  group_by(Pignum, Treatment) %>% filter(Day.oxy > 0) %>%
  ggplot(aes(x=Day.oxy, y=Serum, colour=Treatment)) + geom_smooth() +
  ggtitle("Oxytet concentration in serum over time") + xlab('Time (d)') + ylab('Oxytet concentration (ng/mL)')

weight_oxy %>% group_by(Pignum, Treatment) %>% filter(Day.oxy > 0) %>%
  ggplot(aes(x=Day.oxy, y=Nasal, colour=Treatment)) + geom_smooth() + 
  ggtitle("Oxytet concentration in nasal swabs over time") + xlab('Time (d)') + ylab('Oxytet concentration (ng/mL)')

#this shows negative values for infeed and the inject numbers aren't accurate
weight_oxy %>%  group_by(Pignum, Treatment) %>% filter(Day.oxy > 3) %>%
  ggplot(aes(x=Day.oxy, y=Fecal, colour=Treatment)) + geom_smooth() +
  ggtitle("Oxytet concentration in feces over time") + xlab('Time (d)') + ylab('Oxytet concentration (ng/mL)')

weight_oxy$Treatment <-factor(weight_oxy$Treatment, levels=c('Control', 'Inject', 'InFeed'))
weight_oxy$Day <-factor(weight_oxy$Day, levels=c('4', '7', '14'))

oxy_gathered <- gather(weight_oxy, key=Tissue, value=Concentration, Serum:Fecal)
oxy_gathered$Tissue <- factor(oxy_gathered$Tissue, levels=c("Fecal", "Ileum", "Nasal", "Serum"))

day4 %>% filter(oxy_gathered$Day.oxy > 0) %>% 
  ggplot(aes(x=Day, y=Concentration, group=Tissue, fill=Tissue)) +
  geom_boxplot() + facet_wrap(~Treatment, scales = 'free') + ylab('Concentration (ng/mL)') + 
  geom_jitter(shape = 21, width = .15) +
  ggtitle('Oxytetracylcine concentration by Tissue')
#boxplot not working!!! Probably need it gathered appropriately first
#actually I think the issue is that there are zero's for all tissues without samples


#weight_oxy %>% filter(Treatment=="Inject") %>% filter(Day.oxy > 3) %>%
 # ggplot(aes(x=Day.oxy, y=Fecal)) + geom_smooth() + ggtitle("Oxytet in feces over time for inject")

#weight_oxy %>% filter(Treatment=="Inject") %>% filter(Day.oxy > 3) %>%
 # ggplot(aes(x=Day.oxy, y=Fecal, group=Day.oxy)) + geom_boxplot() + ggtitle("Oxytet in feces over time for inject")
positive<-oxy_gathered %>% filter(Concentration > 0)
#used this in paper draft (oxytet by tissue)
positive %>%  filter(Treatment != "Control") %>%
  ggplot(aes(x=Day, y=Concentration, fill=Tissue)) +
  geom_boxplot() + facet_wrap(~Treatment, scales = 'free') + ylab('Concentration (ng/mL)') + 
  #geom_jitter(shape = 21, width = .15) +
  ggtitle('Oxytetracylcine concentration by Tissue')

Fecal<-positive %>% filter(Tissue=="Fecal")
Fecal %>%  ggplot(aes(x=Day, y=Concentration, group=Treatment, fill=Treatment)) +
  geom_boxplot() + ylab('Concentration (ng/mL)') + 
  geom_jitter(shape = 21, width = .15) +
  ggtitle('Oxytetracycline concentration in Feces')



Fecal_stats<- Fecal %>% group_by(Treatment, Day) %>%
  summarize(mean=mean(Concentration),
            stDev=stDev(Concentration))


#June 11, 2018 - trying to remake oxy figures with correct colours
weight_oxy$Treatment <-factor(weight_oxy$Treatment, levels=c('Control', 'Inject', 'InFeed'))
weight_oxy$Day.oxy<-as.numeric(weight_oxy$Day.oxy)

weight_oxy %>%  group_by(Pignum, Treatment) %>% filter(Day.oxy > 3) %>%
  ggplot(aes(x=Day.oxy, y=Fecal, colour=Treatment)) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm', se=FALSE) +
  ggtitle("Oxytet concentration in feces over time") + xlab('Time (d)') + ylab('Oxytet concentration (ng/mL)')
#this doesn't work because we didn't have fecal measurements for all of these days
#one option for the poster (without se)
weight_oxy %>%  group_by(Pignum, Treatment) %>% filter(Day.oxy > 3 & Day.oxy < 11) %>%
  ggplot(aes(x=Day.oxy, y=Fecal, colour=Treatment)) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm', se=FALSE) +
  ggtitle("Oxytet concentration in feces over time") + xlab('Time (d)') + ylab('Oxytet concentration (ng/mL)')

#or with se
ox_se <- weight_oxy %>%  group_by(Treatment) %>% filter(Day.oxy > 3 & Day.oxy < 11) %>% 
  ggplot(aes(x=Day.oxy, y=Fecal, colour=Treatment)) + geom_smooth(aes(group=Treatment, colour=Treatment), method='lm', se=TRUE) +
  ggtitle("Oxytet concentration in feces over time") + xlab('Time (d)') + ylab('Oxytet concentration (ng/mL)')
#this line prints the current plot that's open but allows you to specify resolution. Note that these dimensions were bigger than I needed!
dev.print(tiff, "oxy_se.tiff", res=800, height=4, width=5, units="in")
#NOTE - also tried filtering out all 0 values but then control disappeared completely

is.factor(weight_oxy$Day.oxy)
summary(weight_oxy)

#Oct2018 - rearranging Bob's wafergen primers file so I can blast the contigs with them

primers<-read.csv("/Users/nicolericker/Documents/DARTE-QM/wafergen_primers.csv", header=TRUE, stringsAsFactors=FALSE, quote="")
warnings()
head(primers)
colnames(primers)<-c("Primer", "Number", "Forward", "Reverse")
head(primers)
primers<-gather(primers, "Forward", "Reverse", key="Direction", value="Primer", na.rm=FALSE)
head(primers)
primers$Primer<-gsub(" ","_", primers$Primer)
head(primers)
primers$Name<-paste(primers$Primer,"_",primers$Number,"_",primers$Direction)
head(primers)
primers$Name<-gsub(" ","", primers$Name)
head(primers)
colnames(primers)<-c("Primer", "Number", "Direction", "Sequence", "Name")
primer_fasta<- primers %>% select(Name, Sequence)
head(primer_fasta)
primer_fasta$Name<-paste(">",primer_fasta$Name)
head(primer_fasta)
primer_fasta$Name<-gsub(" ","", primer_fasta$Name)

write.table(primer_fasta, file="/Users/nicolericker/Documents/DARTE-QM/wafergen_primers.fasta", sep="\t", col.names=FALSE, row.names=FALSE, quote=FALSE)

################# Phyloseq using Jared's pipeline #####################
library(ggplot2)
library(vegan)
library(dplyr)
library(scales)
library(grid)
library(reshape2)
library(phyloseq)

setwd("/Users/nicolericker/Documents/Pig experiments/FY2016/FS1_updated - April 2018- more figs June 2018/")

sharedfile="FS1.V4.shared"    #original name would have ended in opti_mcc.unique_list.shared
taxfile="FS1.V4.taxonomy"     #original name would have ended in opti_mcc.unique_list.0.03.cons.taxonomy

mapfile="FS1.mastermeta.txt"

mothur_data<-import_mothur(mothur_shared_file = sharedfile, 
                           mothur_constaxonomy_file = taxfile)
map<- read.delim(mapfile)

#the column names for the otu_table and the rownames of the map file must match (not order but format)
colnames(otu_table(mothur_data))
rownames(map)

#always check to make sure that they merged properly - have to turn metadata into sample_data
phy_meta<-sample_data(map)
FS1<-merge_phyloseq(mothur_data, phy_meta)
FS1
colnames(tax_table(FS1)) 
#rename these to useful headings

colnames(tax_table(FS1))<-c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")

#now check for unwanted taxa
unwanted<- FS1 %>% subset_taxa(Kingdom != "Bacteria" | 
                                 Family == "mitochondria" |
                                 Class == "Chloroplast")
unwanted  
head(tax_table(unwanted))
head(otu_table(unwanted))
#oops this removed the Archaea - we should keep those; only other was a single chloroplast so we'll just remove that

FS1 <- FS1 %>% subset_taxa(Class != "Chloroplast")
FS1

#save the phyloseq object
saveRDS(FS1, "FS1.NR-analysis-Oct2018.RDS")

### let's remind ourselves of some basic phyloseq options
#from Fan's notes
ntaxa(FS1)   #5792 taxa
sample_variables(FS1)  #16 including tissue, treatment, sow, birth_room, DOB, sex, initial_weight, necropsy.day, pen
rank_names(FS1)  #Kingdom, phylum etc.
sample_names(FS1)  # 312 total X1P59N3AFD4 for example
head(otu_table(FS1))  #otu are row.names and sample_names are columns
sample_data(FS1)  #gives the full metadata - this is already quite tidy
length(tax_table(FS1))   #34753 (lines I presume)
length(otu_table(FS1))  #1807104.. which is much more than the 5792 taxa I was expecting
topNOTUs=names(sort(taxa_sums(FS1), TRUE)[1:100])
head(topNOTUs) #this is just the names of the most abundant OTUs - note they were already sorted apparently
top_100<-prune_taxa(topNOTUs, FS1)
plot_bar(top_100, "treatment")
head(tax_table(top_100))
min(sample_sums(FS1))  #2204 is the minimum
max(sample_sums(FS1))  #91881 is the max
sd(sample_sums(FS1))   #12876.74
summary(tax_table(FS1))
unknown<-subset_taxa(FS1, Kingdom == "unknown")
head(tax_table(unknown)) # 6 taxa - these should be removed
FS1 <- subset_taxa(FS1, Kingdom != "unknown")

#resave the phyloseq object without the unknowns
saveRDS(FS1, "FS1.NR-analysis-Oct2018.RDS")

min(taxa_sums(FS1))
zero<-subset_taxa(FS1, taxa_sums(FS1) == 0)
head(tax_table(zero))
head(otu_table(zero))  

summary(tax_table(FS1))

day0<-subset_samples(FS1, day==0)
length(sample_names(day0))  #52 samples
sample_sums(day0)  #all look quite even
min(sample_sums(day0))  #minimum is 10511
max(sample_sums(day0))  #86069  okay, not that even but still okay
min(taxa_sums(day0)) 
max(taxa_sums(day0))  #82917
ntaxa(day0)  #5769
day0_taxazero<-prune_taxa(taxa_sums(day0)==0, day0)  #creates a new physeq matching this
head(tax_table(day0_taxazero)) #this lets us see which ones were zero
ntaxa(day0)  #still same number therefore not removed.  Use filter_taxa if you want them out
head(otu_table(FS1))
mean(sample_sums(FS1))  #19809.05

mean(sample_sums(day0)) #20802.67
sd(sample_sums(day0))  #10361.56
min(sample_sums(day0))  #10511

day4<-subset_samples(FS1, day==4)
mean(sample_sums(day4)) #22001.82
sd(sample_sums(day4))  #14841.87
min(sample_sums(day4))  #3265

day7<-subset_samples(FS1, day==7)
mean(sample_sums(day7))  #15929.62
sd(sample_sums(day7))  #11889.13
min(sample_sums(day7))  #3138

day11<-subset_samples(FS1, day==11)
mean(sample_sums(day11))  #15229.83
sd(sample_sums(day11))  #3990.111
min(sample_sums(day11))  #3596

day14<-subset_samples(FS1, day==14)
mean(sample_sums(day14))  #21650.86
sd(sample_sums(day14))  #13490.47
min(sample_sums(day14))  #2204

#now I found the command to do this automatically
summary(rowSums(otu_table(FS1)))
summary(sample_sums(otu_table(FS1)))
summary(sample_sums(otu_table(day14)))
summary(sample_sums(otu_table(day11)))
#median similar but day 11 has lower mean and much lower max (20000 vs 90000)
sample_variables(FS1)
fecal<-subset_samples(FS1, tissue=="Feces")
summary(sample_sums(fecal))  #min 3596, max 91880
fecal_day11<-subset_samples(fecal, day==11)
length(sample_names(fecal))  #204 samples
length(sample_names(fecal_day11))  #23 

ileum<-subset_samples(FS1, tissue=="Ileum")
summary(sample_sums(ileum))   #min 2204, max 31890
ileum_day11<-subset_samples(ileum, day==11)  #no samples?

colon<-subset_samples(FS1, tissue=="Colon")
summary(sample_sums(colon))   #min 3138, max 48700
colon_day11<-subset_samples(colon, day==11)  #no samples

plot_bar(fecal, "Phylum", fill="Phylum", facet_grid=day~treatment)
#too many - trim by 2% cutoff before trying that again!

########### Trimming and normalizing the complete FS1 dataset ############

#remove all OTU's that do not occur in at least one sample 
FS1 <- prune_taxa(taxa_sums(FS1) > 0, FS1)
ntaxa(FS1)  #down to 4616 taxa  
summary(otu_table(FS1)) #this isn't very useful
summary(tax_table(FS1))  #checking to make sure the Kingdom unknown were already removed
?rarefy_even_depth  #the phyloseq authors do not recommend this approach
?plot_richness
sample_variables(FS1)
subset_samples(FS1, tissue == "Feces") %>%
  plot_richness(x="day", color="treatment", measures="Shannon") + 
  ggtitle("Fecal Sample Alpha Diversity")
setwd("/Users/nicolericker/Documents/Pig experiments/FY2016")
subset_samples(FS1, tissue == "Colon") %>%
  plot_richness(x="day", color="treatment", scales="free_y") + 
  ggtitle("Colon Sample Alpha Diversity")   #note the overall diversity is similar to fecal (observed mean ~300, shannon mean ~4)

p <- subset_samples(FS1, tissue == "Ileum") %>%
  plot_richness(x="day", color="treatment", scales="free_y") + 
  ggtitle("Ileum Sample Alpha Diversity")   #note that day 7 is greatly reduced, and even for day 4 and 14 the observed is lower (mean ~250)
#from tutorial - how to add boxplots
p + geom_boxplot(data=p$data, aes(x=day, y=value, color=NULL), alpha=0.1)
#doesn't quite work

#exploratory test - heatmap of Proteobacteria
#data("fecal")
#?plot_heatmap
fecal_proteo <- subset_taxa(fecal, Phylum=="Proteobacteria")
f_p_abund <- prune_samples(sample_sums(fecal_proteo) > 200, fecal_proteo)
f_p_abund <- prune_taxa(taxa_sums(f_p_abund) > 10, f_p_abund)
(p <- plot_heatmap(f_p_abund, "NMDS", "bray", "treatment", "Family", sample.order="treatment"))
(p <- plot_heatmap(f_p_abund, "NMDS", "bray", "treatment", "Family", sample.order="day"))

##################### 16S Analysis following Jared's pipeline ##########################
library(phyloseq)
library(tidyverse)
library(vegan)
library(ggpubr) #I couldn't install this due to cowplot missing - look into this
setwd("~/Documents/Pig experiments/FY2016/FS1_updated - April 2018- more figs June 2018/")
#use RDS to load phyloseq object
FS1 <- readRDS("FS1.NR-analysis-Oct2018.RDS")


#### Jared's Functions ####

#Put phyloseq object into a dataframe with an abundance above 0.02% at the phylum level
RelativeAbundanceDf<- function(physeq) {
  physeq %>% tax_glom(taxrank="Phylum") %>% transform_sample_counts(function(x) {
    x/sum(x)
  }) %>% psmelt() %>% filter(Abundance > 0.02) %>% arrange(Phylum)
}

#or glom at the family level (I'm sure we could write a more versatile function)
RelativeAbundanceDf_family<- function(physeq) {
  physeq %>% tax_glom(taxrank="Family") %>% transform_sample_counts(function(x) {
    x/sum(x)
  }) %>% psmelt() %>% filter(Abundance > 0.02) %>% arrange(Family)
}

#Function to plot relative abundance
PlotRelativeAbundance<- function(df) {
  ggplot(df, aes(x=as.factor(day), y = Abundance, fill = Phylum)) +
    facet_grid(treatment ~ .) +
    geom_bar(stat="identity") +
    theme(axis.title.x = element_blank()) + 
    guides(fill=guide_legend(reverse=TRUE, keywidth = 1, keyheight = 1)) +
    ylab("Relative Abundance (Phyla > 2%)") +
    ggtitle("Phylum Composition over Time by Treatment")
}

#Scale reads function to be used prior to ordination
ScaleReads <- function(physeq, n) {
  physeq.scale <- transform_sample_counts(physeq, function(x) {
    (n * x/sum(x))
  })
  otu_table(physeq.scale) <- floor(otu_table(physeq.scale))
  physeq.scale <- prune_taxa(taxa_sums(physeq.scale) > 0, physeq.scale)
  return(physeq.scale)
}

# Function to summarise a data frame and give statistics
DataSummary <- function(data, varname, groupnames) {
  require(plyr)
  SummaryFunc <- function(x, col) {
    c(mean = mean(x[[col]], na.rm = TRUE), sd = sd(x[[col]], na.rm = TRUE))
  }
  data.sum <- ddply(data, groupnames, .fun = SummaryFunc, varname)
  data.sum <- rename(data.sum, c(mean = varname))
}
######### End Functions ###################

#trim phyla at 2% cutoff
FS1.phylum.abundance<- RelativeAbundanceDf(FS1)
colnames(FS1.phylum.abundance)

FS1.family.abundance <- RelativeAbundanceDf_family(FS1)
#Plot and save image
FS1.phylum.fecal.abundance.plot<-FS1.phylum.abundance %>%
  subset(tissue == "Feces") %>%
  PlotRelativeAbundance()

#just plot - can save if you like it
FS1.family.abundance %>% subset(tissue == "Feces") %>%
  PlotRelativeAbundance() + ggtitle("Family Composition in Feces by Treatment over Time")

#plots by count of animals instead of percent of 100 (or 1)

#I may need to work outside of phyloseq to run alpha stats
otu <- import_mothur(mothur_shared_file="/Users/nicolericker/Documents/Pig experiments/FY2016/FS1_updated - April 2018- more figs June 2018/FS1.V4.shared")
 
#or try again from Jared's
min(taxa_sums(fecal))  #let's remove the OTU's that sum to 0
fecal.pruned <- prune_species(speciesSums(fecal) > 0, fecal) #note this should be taxa_sums since speciesSums is deprecated
min(taxa_sums(fecal.pruned))
sample_data(fecal.pruned)$day <- factor(sample_data(fecal.pruned)$day,
                                       levels = c("0", "4", "7", "11", "14"))
richness.data<-plot_richness(fecal.pruned, measures = "Shannon")
richness.df <- richness.data$data        
#could do the same thing for scaled data
sample_variables(fecal)
richness.shannon <- ggplot(richness.df, aes(x=day, y=value), color = treatment) +
  geom_boxplot(aes(color=treatment), position = "dodge") +
  facet_grid(~treatment)
richness.shannon

#looking at scaling function from Jared
min(sample_sums(fecal))  #minimum is 3596
fecal.scale <- ScaleReads(fecal, n = 4000)
#fix the day data into levels
sample_data(fecal.scale)$day <- factor(sample_data(fecal.scale)$day,
                                       levels = c("0", "4", "7", "11", "14"))

#lets look at an ordination
fecal.scale.ordination <- ordinate(physeq=fecal.scale, method = "PCoA", distance = "bray")
#use plot_ordination to plot it and then manipulate in ggplot2

fecal.scale.plot.ordination <- plot_ordination(physeq = fecal.scale, ordination = fecal.scale.ordination,
                                               color = "day", shape = "treatment", title = "PCoA of Fecal Samples") +
  geom_point(aes(color=day), alpha = 0.7, size = 4) +
  geom_point(color = "grey90", size = 1.5)
#the second geom_point makes the center grey for each point

tiff('NR.PCoA_Fecal.tiff', units="in", width=10, height=10, res=300)
fecal.scale.plot.ordination
dev.off()
#there are other ways to do this, just trying Jared's way

#let's plot shannon diversity of the scaled version now
#note that the plot_richness info says to use untrimmed, non-normalized data
richness.scaled.data<-plot_richness(fecal.scale, measures = "Shannon")
richness.scaled.df <- richness.scaled.data$data  
richness.scaled.shannon <- ggplot(richness.scaled.df, aes(x=day, y=value), color=treatment) +
  geom_boxplot(aes(color=treatment), position="dodge") +
  facet_grid(~treatment) + ggtitle("Fecal Diversity") +
  xlab("Day after Treatment") + ylab("Shannon Diversity")
richness.scaled.shannon

#try the ordination on the unscaled
fecal.pruned.ordination <- ordinate(physeq=fecal.pruned, method = "PCoA", distance = "bray")
#use plot_ordination to plot it and then manipulate in ggplot2

fecal.pruned.plot.ordination <- plot_ordination(physeq = fecal.pruned, ordination = fecal.pruned.ordination,
                                                color = "day", shape = "treatment", title = "PCoA of Fecal Samples") +
  geom_point(aes(color=day), alpha = 0.7, size = 4) +
  geom_point(color = "grey90", size = 1.5)

#now to do the stats
#adonis function
richness.scaled.df<-as(sample_data(fecal.scale), "data.frame")
fecal.distance <- distance(fecal.scale, "bray")
fecal.adonis <- adonis(fecal.distance ~ treatment + day, richness.scaled.df)
fecal.adonis
#shows both treatment and day as significant at 0.001.  
#F model for treatment is 2.669; F model for day is 13.207

#check what happens if added in the other order
fecaladonis.reverse <- adonis(fecal.distance ~ day + treatment, richness.scaled.df)
fecaladonis.reverse
#basically identical results

#how about on the unscaled version
richness.df<-as(sample_data(fecal.pruned), "data.frame")
richness.distance<-distance(fecal.pruned, "bray")
richness.adonis <- adonis(richness.distance ~ treatment + day, richness.df)
richness.adonis
#slightly different but barely noticeable; still 0.001 for both
#F model 2.38 for treatment and 11.79 for day


#trying without pruning
distance.fecal<-distance(fecal, "bray")


pen.adonis <-adonis(richness.distance ~ pen + day, richness.df)
pen.adonis
#this also appears quite significant (0.002); 
sow.adonis <- adonis(richness.distance ~ Sow + day, richness.df)
sow.adonis
#slightly significant (0.014)

#use Jared's summary function to look at the other data
sample_data(fecal)$day <- as.factor(sample_data(fecal)$day)
fecal_data<-as.data.frame(sample_data(fecal))
colnames(fecal_data)
birth_weight <- DataSummary(fecal_data, varname="initial_weight", groupnames = "treatment")
birth_weight
#absolutely no difference between groups for birth weight 

#can we do the same analysis by Jules' vegan method?

rownames(fecal_data)  #this corresponds with Jules' group column
share.rare <- rrarefy(otu_table(fecal), min(rowSums(otu_table(fecal))))
rownames(share.rare)  #this is not the same info as the fecal_data so I can't combine
#I need to have the OTU's as columns to use vegan
shared<-read.table("FS1_updated - April 2018- more figs June 2018/FS1.V4.shared", header = TRUE)
rownames(shared)<-shared[,2]
shared <- shared[,-c(1,2,3)]
shared_fecal<-shared[rownames(shared) %in% rownames(fecal_data),]
tax.df<-data.frame(tax_table(FS1))
shared<-shared[rowSums(shared) > 2000,]  #removes samples with less than 2000 reads
shared <- shared[,colSums(shared) > 10]  #removes OTU's occurring less than 10 times globally
shared_fecal<-shared[rownames(shared) %in% rownames(fecal_data),]
#204 samples

min(rowSums(shared_fecal))
fecal.share.rare<-(rrarefy(shared_fecal, min(rowSums(shared_fecal))))  #rarified for alpha and beta diversity calculations
rownames(fecal.share.rare) == rownames(fecal_data)  #making sure they're in the same order
fecal.share.rare.original <- fecal.share.rare  #for boxplots later

rowSums(fecal.share.rare)
fecal_data$shannon <- diversity(fecal.share.rare)
fecal_data %>% ggplot(aes(x=day, y=shannon, group=set, fill=treatment)) +
  geom_boxplot() + ggtitle('Alpha diversity') + ylab("Shannon Diversity Index")

#let's do the full dataset 
metadata<-as.data.frame(sample_data(FS1))
share.rare <- rrarefy(shared, min(rowSums(shared)))
rownames(metadata)
rownames(share.rare)
share.rare<-shared[rownames(share.rare) %in% rownames(metadata)]
rownames(share.rare) == rownames(metadata) #not in the same order this time
share.rare <-share.rare[match(rownames(metadata), rownames(share.rare)),]
rownames(share.rare) == rownames(metadata)  #now in the the same order
share.rare.original <- share.rare #for boxplots later

metadata$shannon <- diversity(share.rare)

metadata %>% filter(tissue=="Ileum") %>%
  ggplot(aes(x=day, y=shannon, group = set, fill=treatment)) + geom_boxplot() + ggtitle("Ileum Alpha Diversity") + ylab("Shannon Diversity Index")
#group=set is why they aren't all combined as a single value per day
#set is day_tissue_treatment

metadata %>% filter(tissue=="Colon") %>%
  ggplot(aes(x=day, y=shannon, group=set, fill=treatment)) + geom_boxplot() + ggtitle("Colon Alpha Diversity") + ylab("Shannon Diversity Index")

######## Pairwise Adonis ##############
library(funfuns)
PWadon <- pairwise.adonis(share.rare, metadata$set)

fecesvsfeces<-grep(".*_Feces_.* vs .*_Feces_*", PWadon$pairs)
PWadon.feces <-PWadon[fecesvsfeces,]
#but this compares everything (including things that don't make sense
D0.feces <- grep("D0_Feces_.* vs D0_Feces_.*", PWadon$pairs)
D4.feces <- grep("D4_Feces_.* vs D4_Feces_.*", PWadon$pairs)
D7.feces <- grep("D7_Feces_.* vs D7_Feces_.*", PWadon$pairs)
D11.feces <- grep("D11_Feces_.* vs D11_Feces_.*", PWadon$pairs)
D14.feces <- grep("D14_Feces_.* vs D14_Feces_.*", PWadon$pairs)
PWadon.good.feces<-PWadon[c(D0.feces, D4.feces, D7.feces, D11.feces, D14.feces),]
write.table(PWadon.good.feces,file="NR-FS1_PWadon_feces.txt", quote=FALSE, row.names=FALSE, sep="\t")


D4.colon <- grep("D4_Colon_.* vs D4_Colon_.*", PWadon$pairs)
D7.colon <- grep("D7_Colon_.* vs D7_Colon_.*", PWadon$pairs)
D14.colon <- grep("D14_Colon_.* vs D14_Colon_.*", PWadon$pairs)
PWadon.good.colon<-PWadon[c(D4.colon, D7.colon, D14.colon),]
write.table(PWadon.good.colon,file="NR-FS1_PWadon_colon.txt", quote=FALSE, row.names=FALSE, sep="\t")
#note that my numbers don't match Jules....

#plot these relative to the control
PWadon.fecaltocont <- PWadon.good.feces[grep('control', PWadon.good.feces$pairs),] 

PWadon.fecaltocont$day<-gsub('D([0-9]+)_.*', '\\1', PWadon.fecaltocont$pairs)
PWadon.fecaltocont$treatment<-rep(c("inject", "oral"), 5)
#PWadon.fecaltocont$group<-gsub('D[0-9]+_Feces_control vs D[0-9]+_Feces_[A-Za-z]+)', '\\1', PWadon.fecaltocont$pairs)
PWadon.fecaltocont$day<-factor(PWadon.fecaltocont$day, levels=c("0", "4", "7", "11", "14"))

pAdon.fecal<-ggplot(PWadon.fecaltocont, aes(x=day, y=F.Model, color=treatment)) +
  geom_path(aes(group=treatment), size=1.3) +
  scale_fill_manual(values=c('#00BA38', '#619CFF')) +
  geom_vline(xintercept=3, color='purple', size=1, alpha=0.75) +
  geom_label(aes(label=p.value, fill=treatment), color='black', show.legend=FALSE) +
  scale_color_manual(values=c("#00BA38", "#619CFF"))
#looks great, but very different numbers from Jules' 

#let's check the colon ones
PWadon.colontocont <- PWadon.good.colon[grep('control', PWadon.good.colon$pairs),] 

PWadon.colontocont$day<-gsub('D([0-9]+)_.*', '\\1', PWadon.colontocont$pairs)
PWadon.colontocont$treatment<-rep(c("inject", "oral"), 3)
#PWadon.fecaltocont$group<-gsub('D[0-9]+_Feces_control vs D[0-9]+_Feces_[A-Za-z]+)', '\\1', PWadon.fecaltocont$pairs)
PWadon.colontocont$day<-factor(PWadon.colontocont$day, levels=c("4", "7", "14"))

pAdon.colon<-ggplot(PWadon.colontocont, aes(x=day, y=F.Model, color=treatment)) +
  geom_path(aes(group=treatment), size=1.3) +
  scale_fill_manual(values=c('#00BA38', '#619CFF')) +
  geom_vline(xintercept=2, color='purple', size=1, alpha=0.75) +
  geom_label(aes(label=p.value, fill=treatment), color='black', show.legend=FALSE) +
  scale_color_manual(values=c("#00BA38", "#619CFF"))

################### NMDS Ordination ###############
rownames(metadata) == rownames(share.rare)
rownames(fecal_data) == rownames(shared_fecal)
 #using Jules' NMDS ellipse function
NMDS <- NMDS_ellipse(metadata=fecal_data, OTU_table=shared_fecal, grouping_set='set')


#this didn't work

############ Ellipses in Phyloseq ###############
data(enterotype)
# simplify the data
ord = ordinate(fecal.pruned, formula = ~treatment, "NMDS", "bray")
(ordplot <- plot_ordination(fecal.pruned, ord, "samples", color="treatment",shape="day"))
#Run 20 stress 0.164986

#this shows ellipses of two types - t-distribution or normal distribution 
ordplot + 
  stat_ellipse(type = "norm", linetype = 2) +
  stat_ellipse(type = "t") +
  theme_bw()
#oops - too confusing with all the days there
ord.d0=ordinate(day0, formula = ~treatment, "NMDS", "bray")
#no solution reached
prd.day4= ordinate(day4, formula = ~treatment, "NMDS", "bray")
(ordplot2 <- plot_ordination(day4, prd.day4, "samples", color="treatment", shape="day"))
sample_data(day4)$day <- factor(sample_data(day4)$day, levels = c("4"))
ordplot2 + 
  stat_ellipse(type = "t") +
   theme_bw()

sample_data(day7)$day <- factor(sample_data(day7)$day, levels = c("7"))
ord.day7= ordinate(day7, formula = ~treatment, "NMDS", "bray")
(ordplot3 <- plot_ordination(day7, ord.day7, "samples", color="treatment", shape="day"))
unique(sample_data(day7)$tissue)
ordplot3 + 
  stat_ellipse(type = "t") +
  theme_bw()

#oops - I've included all tissues instead of just fecals
#therefore I know how to do it now and will focus on writing for the time being :)


#Trying Adonis Phil's way
bc_all=phyloseq::distance(fecal, method="bray")
all_df = as(sample_data(fecal), "data.frame")
adonis_all <- adonis(bc_all ~ day + treatment + pen, data = all_df)
adonis_all
#Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)    
#day         1     5.270  5.2702 22.3116 0.09821  0.001 ***
#  treatment   2     1.005  0.5027  2.1284 0.01874  0.002 ** 
#  pen         3     0.852  0.2840  1.2023 0.01588  0.109    
#Residuals 197    46.533  0.2362         0.86717           
#Total     203    53.661                 1.00000 
adonis_weight <- adonis(bc_all ~ day + treatment + initial_weight, data = all_df)
adonis_weight
#Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)    
#day              1     5.270  5.2702 22.3547 0.09821  0.001 ***
#  treatment        2     1.005  0.5027  2.1325 0.01874  0.002 ** 
#  initial_weight   1     0.470  0.4701  1.9940 0.00876  0.008 ** 
#  Residuals      199    46.915  0.2358         0.87429           
#Total          203    53.661                 1.00000 

adonis_sow <- adonis(bc_all ~ day + Sow, data = all_df)
adonis_sow
#also significant : 0.018

adonis_room<- adonis(bc_all ~ day + room, data = all_df)
adonis_room
#also significant : 0.001

adonis_many <- adonis(bc_all ~ day + treatment + Sow + room + initial_weight + pen + Sex + Necropsy.day, data=all_df)
adonis_many
Call:
  adonis(formula = bc_all ~ day + treatment + Sow + room + initial_weight +      pen + Sex + Necropsy.day, data = all_df) 

#Permutation: free
#Number of permutations: 999

#Terms added sequentially (first to last)

#Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)    
#day              1     5.270  5.2702 22.7502 0.09821  0.001 ***
#  treatment        2     1.005  0.5027  2.1702 0.01874  0.001 ***
#  Sow              1     0.454  0.4539  1.9592 0.00846  0.018 *  
#  initial_weight   1     0.474  0.4743  2.0475 0.00884  0.007 ** 
#  pen              3     0.847  0.2825  1.2194 0.01579  0.092 .  
#Sex              1     0.463  0.4628  1.9980 0.00863  0.007 ** 
#  Necropsy.day     1     0.437  0.4372  1.8875 0.00815  0.020 *  
#  Residuals      193    44.710  0.2317         0.83319           
#Total          203    53.661                 1.00000           
#---
#  Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#Phil's pairwise adonis function
pairwise.adonis <- function(x,factors, sim.function = 'vegdist', sim.method = 'bray', p.adjust.m ='bonferroni')
{
  library(vegan)
  
  co = combn(unique(as.character(factors)),2)
  pairs = c()
  F.Model =c()
  R2 = c()
  p.value = c()
  
  for(elem in 1:ncol(co)){
    if(sim.function == 'daisy'){
      library(cluster); x1 = daisy(x[factors %in% c(co[1,elem],co[2,elem]),],metric=sim.method)
    } else{x1 = vegdist(x[factors %in% c(co[1,elem],co[2,elem]),],method=sim.method)}
    
    ad = adonis(x1 ~ factors[factors %in% c(co[1,elem],co[2,elem])] );
    pairs = c(pairs,paste(co[1,elem],'vs',co[2,elem]));
    F.Model =c(F.Model,ad$aov.tab[1,4]);
    R2 = c(R2,ad$aov.tab[1,5]);
    p.value = c(p.value,ad$aov.tab[1,6])
  }
  p.adjusted = p.adjust(p.value,method=p.adjust.m)
  sig = c(rep('',length(p.adjusted)))
  sig[p.adjusted <= 0.05] <-'.'
  sig[p.adjusted <= 0.01] <-'*'
  sig[p.adjusted <= 0.001] <-'**'
  sig[p.adjusted <= 0.0001] <-'***'
  
  pairw.res = data.frame(pairs,F.Model,R2,p.value,p.adjusted,sig)
  print("Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1")
  return(pairw.res)
} 

FS1_feces_df = as.data.frame(otu_table(fecal))
FS1_feces_df_t = t(FS1_feces_df)
sample_data_feces = sample_data(fecal)
pairwise.adonis(x=FS1_feces_df_t,factors=sample_data_feces$Sex,sim.function='vegdist',sim.method='bray',p.adjust.m='bonferroni')

#realized that the pigs are all in there multiple times since fecal samples are taken on different days
#therefore these could seem significant when they aren't?  
#trying just the day 7
fecal_day7<-subset_samples(fecal, day==7)
bc_fecal_day7=phyloseq::distance(fecal_day7, method="bray")
fecal_day7_df = as(sample_data(fecal_day7), "data.frame")
adonis_fecal_day7 <- adonis(bc_fecal_day7 ~ treatment + pen + Sow + initial_weight + Sex + Necropsy.day, data = fecal_day7_df)
adonis_fecal_day7
#that's better except that pen effect is now significant 
#Call:
 # adonis(formula = bc_fecal_day7 ~ day + treatment + pen + Sow +      initial_weight + Sex + Necropsy.day, data = fecal_day7_df) 

#Permutation: free
#Number of permutations: 999

#Terms added sequentially (first to last)

#Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)    
#treatment       2    0.9250 0.46251  2.3951 0.09857  0.001 ***
#  pen             3    0.9297 0.30989  1.6048 0.09906  0.001 ***
#  Sow             1    0.2890 0.28899  1.4966 0.03079  0.056 .  
#initial_weight  1    0.2060 0.20595  1.0665 0.02195  0.338    
#Sex             1    0.2648 0.26478  1.3711 0.02821  0.085 .  
#Necropsy.day    1    0.2048 0.20482  1.0607 0.02182  0.349    
#Residuals      34    6.5656 0.19311         0.69960           
#Total          43    9.3849                 1.00000 

adonis_fd7_sow <- adonis(bc_fecal_day7 ~ treatment + Sow, data = fecal_day7_df)
adonis_fd7_sow
adonis_fd7_pen <- adonis(bc_fecal_day7 ~ treatment + pen, data = fecal_day7_df)
adonis_fd7_pen


FS1_fd7_df = as.data.frame(otu_table(fecal_day7))
FS1_fd7_df_t = t(FS1_fd7_df)
sample_data_fd7 = sample_data(fecal_day7)
pairwise.adonis(x=FS1_fd7_df_t,factors=sample_data_fd7$pen,sim.function='vegdist',sim.method='bray',p.adjust.m='bonferroni')
#[1] "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1"
#pairs  F.Model         R2 p.value p.adjusted sig
#1  1A vs 1B 1.642941 0.11220018   0.033      0.495    
#2  1A vs 2A 1.386253 0.10355793   0.120      1.000    
#3  1A vs 2B 2.433643 0.15768430   0.005      0.075    
#4  1A vs 3A 2.396505 0.17889029   0.006      0.090    
#5  1A vs 3B 4.330197 0.24986425   0.001      0.015   .
#6  1B vs 2A 1.247826 0.08758010   0.179      1.000    
#7  1B vs 2B 1.248083 0.08185181   0.172      1.000    
#8  1B vs 3A 1.686387 0.12321635   0.021      0.315    
#9  1B vs 3B 2.318504 0.14207821   0.001      0.015   .
#10 2A vs 2B 1.254519 0.08800852   0.186      1.000    
#11 2A vs 3A 1.628627 0.12896312   0.031      0.465    
#12 2A vs 3B 2.127463 0.14063578   0.003      0.045   .
#13 2B vs 3A 1.443397 0.10736845   0.096      1.000    
#14 2B vs 3B 1.756602 0.11148353   0.011      0.165    
#15 3A vs 3B 1.773237 0.12874514   0.003      0.045   .
#Note that there is a significant difference for 3A vs 3B?  
#Perhaps we should leave pen effect out of this study since we have a separate study looking at this and we had small pens in the same room for this one
pairwise.adonis(x=FS1_fd7_df_t,factors=sample_data_fd7$Sow,sim.function='vegdist',sim.method='bray',p.adjust.m='bonferroni')
#nothing significant in the sows suggesting that birth mother was not a driving factor on day7 at least


#combined figure attempt October 30, 2018
p1<- all_oxy %>% filter(Day.oxy==1 | Day.oxy==4) %>% mutate(Treatment_day=paste(Treatment, Day.oxy, sep=" Day ")) %>%
  ggplot(aes(x=Weight, y=Serum, colour=Treatment_day)) + geom_point(size=0.7, alpha=0.5) + 
  geom_smooth(aes(group=Treatment_day, colour=Treatment_day), method='lm') + 
  scale_colour_brewer(palette="Dark2") +
  xlab('Weight on day 0 (kg)') + ylab('Plasma concentration (ng/mL)') 
#saved as Oxy concentrations by weight Dark2 colors.jpeg ###############################

saveRDS(all_oxy, file="all_oxy.RDS")  # this saves the object as it is here - shared with Schuyler for advice
#from Schyler
ggplot(all_oxy, aes(x=Weight, y=Serum, colour=Treatment)) + 
  geom_point(data = subset(all_oxy, Day.oxy==1 | Day.oxy==4), size=0.7, alpha=0.5) +
  geom_smooth(data = subset(all_oxy, Day.oxy==1), aes(group=Treatment, colour=Treatment, linetype="dashed"), method='lm') +
  geom_smooth(data = subset(all_oxy, Day.oxy==4), aes(group=Treatment, colour=Treatment, linetype="solid"), method='lm') +
  xlab('Weight on day 0 (kg)') + ylab('Plasma concentration (ng/mL)') + theme(legend.key = element_text(Day.oxy))
#the legend was only saying 'dashed' or 'solid' so I kept working on the one above this; note that the lines seemed to be with the wrong subsets when I ran this command


#also need a summary of oxytet by Tissue
head(all_oxy)
range(all_oxy$Serum)
range(all_oxy$Fecal)

oxy_summary<-all_oxy %>% group_by(Treatment, Day.oxy) %>% summarise(mean_Plasma=mean(Serum), 
                                                                    mean_Nasal=mean(Nasal),
                                                                    mean_Ileum=mean(Ileum),
                                                                    mean_Fecal=mean(Fecal))
write.table(oxy_summary, file="oxy_summary.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names=TRUE)
mean_treatments<-all_oxy %>% group_by(Treatment) %>% summarise(Plasma=mean(Serum),
                                                    Nasal=mean(Nasal),
                                                    Ileum=mean(Ileum),
                                                    Fecal=mean(Fecal))
write.table(mean_treatments, file="mean_oxy_treatments.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names=TRUE)

oxy_plasma <- oxy_summary %>% select("Treatment", "Day.oxy", "mean_Plasma") %>% spread(key=Day.oxy, value=mean_Plasma)
View(oxy_plasma)
colnames(oxy_plasma)[2:9]<-paste('Day', colnames(oxy_plasma)[2:9], sep=" ")
oxy_plasma <- oxy_plasma %>% select("Treatment", "Day 0", "Day 1", "Day 3", "Day 4", "Day 7", "Day 11", "Day 14") 
oxy_plasma$`Day 0` <- format(round(oxy_plasma$`Day 0`, digits=3),scientific=TRUE)
write.table(oxy_plasma, file="oxy_plasma_summary.tsv", sep="\t", row.names=FALSE, col.names=TRUE, quote=FALSE)

oxy_plasma_sd<-all_oxy %>% group_by(Treatment, Day.oxy) %>% summarise(sd_Plasma=sd(Serum))
oxy_plasma_sd <- oxy_plasma_sd %>% spread(key=Day.oxy, value=sd_Plasma)
dim(oxy_plasma_sd)
colnames(oxy_plasma_sd)[2:9]<-paste('sd_Day', colnames(oxy_plasma_sd)[2:9], sep=" ")
oxy_plasma_sd <- oxy_plasma_sd %>% select("Treatment", "sd_Day 0", "sd_Day 1", "sd_Day 3", "sd_Day 4", "sd_Day 7", "sd_Day 11", "sd_Day 14") 
oxy_plasma_combined<-merge(oxy_plasma, oxy_plasma_sd, by = "Treatment")
oxy_plasma_combined<-unite(oxy_plasma_combined, "Day 0", "sd_Day 0", col = "Day 0", sep=" +/- ")
oxy_plasma_combined<-unite(oxy_plasma_combined, "Day 1", "sd_Day 1", col = "Day 1", sep=" +/- ")
oxy_plasma_combined<-unite(oxy_plasma_combined, "Day 3", "sd_Day 3", col = "Day 3", sep=" +/- ")
oxy_plasma_combined<-unite(oxy_plasma_combined, "Day 4", "sd_Day 4", col = "Day 4", sep=" +/- ")
oxy_plasma_combined<-unite(oxy_plasma_combined, "Day 7", "sd_Day 7", col = "Day 7", sep=" +/- ")
oxy_plasma_combined<-unite(oxy_plasma_combined, "Day 11", "sd_Day 11", col = "Day 11", sep=" +/- ")
oxy_plasma_combined<-unite(oxy_plasma_combined, "Day 14", "sd_Day 14", col = "Day 14", sep=" +/- ")
View(oxy_plasma_combined)
oxy_plasma_t<-t(oxy_plasma_combined)
write.table(oxy_plasma_combined, file="Oxy_plasma_with_sd.tsv", sep="\t", quote=FALSE, row.names=FALSE, col.names=TRUE)
write.table(oxy_plasma_t, file="transposed_Oxy_plasma_with_sd.tsv", sep="\t", quote=FALSE, row.names=TRUE, col.names=FALSE)
View(oxy_plasma_t)


day4_summary<-all_oxy %>% filter(Day.oxy == 4) %>% group_by(Treatment) %>% 
  summarise(mean_Plasma=mean(Serum), 
            mean_Nasal=mean(Nasal),
            mean_Ileum=mean(Ileum),
            mean_Fecal=mean(Fecal))
day4_summary$mean_Plasma<-round(day4_summary$mean_Plasma, digits=2)
day4_summary$mean_Nasal<-round(day4_summary$mean_Nasal, digits=2)
day4_summary$mean_Ileum<-round(day4_summary$mean_Ileum, digits=2)
day4_summary$mean_Fecal<-round(day4_summary$mean_Fecal, digits=2)


write.table(day4_summary, file="Oxy_summary_day4.tsv", sep="\t", quote=FALSE, row.names=FALSE, col.names=TRUE)

day4_sd<-all_oxy %>% filter(Day.oxy == 4) %>% group_by(Treatment) %>% 
  summarise(sd_Plasma=sd(Serum), 
            sd_Nasal=sd(Nasal),
            sd_Ileum=sd(Ileum),
            sd_Fecal=sd(Fecal))
day4_sd$sd_Plasma<-round(day4_sd$sd_Plasma, digits=2)
day4_sd$sd_Nasal<-round(day4_sd$sd_Nasal, digits=2)
day4_sd$sd_Ileum<-round(day4_sd$sd_Ileum, digits=2)
day4_sd$sd_Fecal<-round(day4_sd$sd_Fecal, digits=2)
write.table(day4_sd, file="Oxy_summary_day4_stdev.tsv", quote=FALSE, sep="\t", row.names=FALSE, col.names=TRUE)

oxy_day4<-merge(day4_summary, day4_sd, by = "Treatment")
oxy_day4<-unite(oxy_day4, "mean_Plasma", "sd_Plasma", col = "Plasma", sep=" +/- ")
oxy_day4<-unite(oxy_day4, "mean_Nasal", "sd_Nasal", col = "Nasal", sep=" +/- ")
oxy_day4<-unite(oxy_day4, "mean_Ileum", "sd_Ileum", col = "Ileum", sep=" +/- ")
oxy_day4<-unite(oxy_day4, "mean_Fecal", "sd_Fecal", col = "Fecal", sep=" +/- ")

write.table(oxy_day4, file="Oxy_summary_day4_withstdev.tsv", sep="\t", row.names=FALSE, col.names=TRUE, quote=FALSE)
##################
